### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by szg251**
Updated some of the dependencies:
- easy-ps
- plutip (transitively haskell-nix, hackage-nix, iohk-nix and CHaP)
- blockfrost
- ogmios 5.6 -> 6; response format changed

### Blocked by
https://github.com/mlabs-haskell/plutip/pull/183

### Pre-review checklist

- [x] All code has been formatted using our config (`make format`)
- [x] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

**Comment by szg251** [comment]
Plutip's `local-cluster` seems to be working correctly, spinning up the cluster with tada distributed, and keys generated.
I could also verify that `plutip-server` is working, and the cluster can be started with the `/start` endpoint.
I still have a problem with CTL: it can start the cluster, prepare the wallets, and query them, but it fails with a timeout when it's balancing a tx (before submitting a tx)

**Comment by szg251** [comment]
Unfortunately I didn't have enough time to finish this, so I pushed my last results to `0fd1270`

As discussed in a thread [here](https://mlabs-corp.slack.com/archives/C02QXGRGAPK/p1695114397088609), the latest versions of cardano-node (8.0.0 and above) require a new version of Ogmios (6.0). This version introduces a lot of breaking changes: most notably, they're using a standard JSON-RPC 2.0 scheme for the websocket messages. There's a beautiful migration guide here:
https://github.com/CardanoSolutions/ogmios/blob/master/architectural-decisions/accepted/017-api-version-6-major-rewrite.md#migration-guide

Last status:
- renamed `JsonWsp` modules, data types etc to `JsonRpc2`
- modified types and codecs in `JsonRpc2` module to comply with the standard
- updated ogmios most calls (see list below) in [/src/Internal/QueryM/Ogmios.purs](https://github.com/Plutonomicon/cardano-transaction-lib/blob/0fd1270bc21d5276f2d2b9d94de820e7c8c80af9/src/Internal/QueryM/Ogmios.purs)
- updated most ogmios request and response types in the same module
- updated protocol parameters data type
- updated test schema in [/fixtures/schemata/JsonRpc2/UtxoQueryResponse.medea](https://github.com/Plutonomicon/cardano-transaction-lib/blob/0fd1270bc21d5276f2d2b9d94de820e7c8c80af9/fixtures/schemata/JsonRpc2/UtxoQueryResponse.medea)
- updated generate fixtures file

What's left:
- finishing call type updates (method names are updated where I could find an equivalent in the new version)
- verifying/updating error responses
- finding out the story with `stakePool` queries: it looks like there's no `poolIds` query anymore, so a few things might have to change here

| call | status |
|---|---|
| querySystemStartCall |request, success response|
| queryCurrentEpochCall |request, success response|
| queryEraSummariesCall |request, success response|
| queryProtocolParameters |request, success response|
| queryChainTipCall |request, success response|
| queryPoolIdsCall -> queryStakePoolsCall  | method name only |
| queryPoolParametersCall  | - |
| queryDelegationsAndRewards  | method name only |
| submitTxCall |request, success response|
| evaluateTxCall |request, success response|
| acquireMempoolSnapshotCall | method name only |
| mempoolSnapshotHasTxCall | method name only |
| mempoolSnapshotNextTxCall | method name only |
| mempoolSnpashotSizeAndCapacityCall | method name only |
| releaseMempoolCall | method name only |

**Comment by zmrocze** [comment]
Thanks @szg251 for the status comment. Definitely helped to finish the pr.

**Comment by zmrocze** [comment]
@klntsky some easily missable details:

 - I removed tests concerning Ogmios Utxo query. It was obsolete already before, but went unnoticed. If you could double check these are unneeded.
 - removed Ogmios Utxo query fixtures and json schema
 - removed Base64 module - not needed as mempool tx is hex encoded now
 - minor changes to how decoding errors are handled - here im pretty sure behaviour is unchanged

**Review by klntsky — APPROVED** [review]

File: flake.nix [hunk 1]
```diff
@@ -4,29 +4,57 @@
...
+
+    ogmios = {
+      url = "github:CardanoSolutions/ogmios/v6.0.0";
       flake = false;
     };
 
-    # ogmios nixos module (remove and replace with the above after merging and updating)
-    ogmios-nixos.url = "github:mlabs-haskell/ogmios";
+    kupo-nixos = {
+      url = "github:mlabs-haskell/kupo-nixos/6f89cbcc359893a2aea14dd380f9a45e04c6aa67";
```
Comment (aciceri) [inline flake.nix:50]: We can remove the revision I believe.

File: flake.nix [hunk 2]
```diff
@@ -35,25 +63,23 @@
...
-    haskell-nix.url = "github:mlabs-haskell/haskell.nix";
-    iohk-nix = {
-      follows = "plutip/iohk-nix";
-      flake = false;
-    };
-    CHaP = {
-      url = "github:input-output-hk/cardano-haskell-packages?ref=repo";
-      flake = false;
+    plutip = {
+      url = "github:mlabs-haskell/plutip/1bf0b547cd3689c727586abb8385c008fb2a3d1c";
```
Comment (aciceri) [inline flake.nix:75]: Is there any reason why we fix the revision? @szg251

File: flake.nix (hunk 1 above)
Comment (aciceri) [reply flake.nix:50]: @szg251 right?

**Pull request merged**